common/jenkins-agents/nodejs12/docker/Dockerfile

```diff
@@ -1,52 +1,87 @@
+    io.openshift.tags="openshift,jenkins,agent,nodejs"
+
+ARG nexusUrl
+ARG nexusAuth
+
+ENV NODEJS_VERSION=12 \
+    NPM_CONFIG_PREFIX=$HOME/.npm-global \
+    PATH=$HOME/node_modules/.bin/:$HOME/.npm-global/bin/:$PATH \
+    BASH_ENV=/usr/local/bin/scl_enable \
+    ENV=/usr/local/bin/scl_enable \
+    PROMPT_COMMAND=". /usr/local/bin/scl_enable" \
+    LANG=en_US.UTF-8 \
+    LC_ALL=en_US.UTF-8
+
+# install google-chrome (for angular)
+RUN yum-config-manager --enable rhel-7-server-extras-rpms && \
+    yum-config-manager --enable rhel-7-server-optional-rpms && \
+    yum makecache
+
+ADD https://dl.google.com/linux/direct/google-chrome-stable_current_x86_64.rpm /root/google-chrome-stable_current_x86_64.rpm
+RUN yum -y install redhat-lsb libXScrnSaver
+RUN yum -y install /root/google-chrome-stable_current_x86_64.rpm && \
+    ln -s /usr/lib64/libOSMesa.so.8 /opt/google/chrome/libosmesa.so && \
+    yum clean all && \
+    dbus-uuidgen > /etc/machine-id
+
+
+# Install cypress dependencies
+# Please note: xorg-x11-server-Xvfb is not available on RHEL via yum anymore, so "RUN yum install -y xorg-x11-server-Xvfb" won't work.
+#   Therefore this Dockerfile uses the version from CentOS instead.
+ADD http://mirror.centos.org/centos/7/os/x86_64/Packages/xorg-x11-server-Xvfb-1.20.4-10.el7.x86_64.rpm /root/xorg-x11-server-Xvfb.x86_64.rpm
+RUN yum install -y /root/xorg-x11-server-Xvfb.x86_64.rpm && \
+    yum install -y gtk2-2.24* && \
+    yum install -y libXtst* && \
+    yum install -y libXScrnSaver* && \
+    yum install -y GConf2* && \
+    yum install -y alsa-lib* && \
+    yum install -y nss-devel libnotify-devel gnu-free-sans-fonts && \
+    yum clean all -y
+# libXScrnSaver* provides libXss
+# GConf2* provides libgconf-2
+# alsa-lib* provides libasound
+
+
+# Install NodeJS + Yarn + Angular CLI
+# installation details see: https://www.softwarecollections.org/en/
+# and the base image relies on scl_enable
+COPY contrib/bin/scl_enable /usr/local/bin/scl_enable
+
+# Install NodeJS
+# https://github.com/nodesource/distributions#installation-instructions-1
+RUN curl --silent --location https://rpm.nodesource.com/setup_12.x | bash - && \
+    yum install -y nodejs && \
+    yum install -y gcc-c++ make
+
+# Install Yarn
+# https://classic.yarnpkg.com/en/docs/install/#centos-stable
+RUN curl --silent --location https://dl.yarnpkg.com/rpm/yarn.repo | tee /etc/yum.repos.d/yarn.repo && \
+    yum install -y yarn && \
+    yum clean all -y
+
+RUN npm config set registry=$nexusUrl/repository/npmjs/ && \
+    npm config set always-auth=true && \
+    npm config set _auth=$(echo -n $nexusAuth | base64) && \
+    npm config set email=no-reply@opendevstack.org && \
+    npm config set ca=null && \
+    npm config set strict-ssl=false && \
+    yarn config set registry $nexusUrl/repository/npmjs/ -g && \
+    npm install -g @angular/cli@8.0.1 --unsafe-perm=true --allow-root && \
+    npm install -g cypress@3.3.1 --unsafe-perm=true --allow-root > /dev/null && \

```

Comment:
@cschweikert can we drop installing `@angular/cli@8.0.1` and `cypress@3.3.1`? That would be ideal because we can then get GitHub Actions to work. Further, it feels wrong for me to back versions of these tools into the agent.

Reply:
* ``@angular/cli@8.0.1``: We can try to remove it. It is installed now via Jenkinsfile during provisioning: https://github.com/opendevstack/ods-quickstarters/blob/2c4514d411c7cabedda191f94fcf6a0798530e37/fe-angular/Jenkinsfile#L13
* ``cypress@3.3.1``: Having this in was nice, when a build should run e2e tests without prior installation of the cypress testing framework. This usually takes about one minute. But then again you are limited to this specific version. There are other ways of baking cypress binaries into a custom agent image. Last time I did it with a ``Dockerfile`` like this:

```
# This FROM tag is informational. It is overwritten by OpenShift (see oc-template.yaml file)
FROM cd/jenkins-slave-nodejs10-angular:2.x

USER root

# add cypress@4.5.0 binaries to /home/jenkins/.cache/Cypress/4.5.0/Cypress/...
# this allows to disable post-installation of cypress@4.5.0 by setting the environment variable
#   CYPRESS_INSTALL_BINARY=0
# cypress@4.5.0 can use those binaries by setting environment variable to the cache folder that holds the binaries
# for the respective versions of cypress
#   CYPRESS_CACHE_FOLDER=/home/jenkins/.cache/Cypress
ADD https://download.cypress.io/desktop/4.5.0?platform=linux&arch=x64 /home/jenkins/cypress/cypress-linux.zip
RUN md5sum /home/jenkins/cypress/cypress-linux.zip && \
    echo "26835296a4edb1dea35795f59213c519 /home/jenkins/cypress/cypress-linux.zip" | md5sum -c && \
    mkdir -p /home/jenkins/.cache/Cypress/4.5.0 && \
    unzip -q /home/jenkins/cypress/cypress-linux.zip -d /home/jenkins/.cache/Cypress/4.5.0

# fix access rights
RUN chgrp -R 0 $HOME && \
    chmod -R g=u $HOME && \
    chmod ug=rx /home/jenkins/.cache/Cypress/4.5.0/Cypress/Cypress
USER 1001
```

This will allow skipping the post-installation step of cypress binaries (faster build, no issues with proxy servers, adds missing file consistancy check (MD5) for the binaries, etc.)

tl;dr

Yes, let's drop installing ``@angular/cli@8.0.1`` and ``cypress@3.3.1``.

Reply:
Cool. Yes I'm in favour of that approach. We should have a section in the docs with "recipes" like this one.